neusomatic/python/preprocess.py

```diff
@@ -329,10 +329,65 @@ def preprocess(work, mode, reference, region_bed, tumor_bam, normal_bam, dbsnp,
             if os.path.exists(work_dataset_split):
                 shutil.rmtree(work_dataset_split)
             os.mkdir(work_dataset_split)
+            ensemble_bed_i = ensemble_beds[i] if ensemble_tsv else None
+            if add_extra_features:
+                work_tumor_i = os.path.dirname(filtered_vcf)
+                extra_features_tsv = os.path.join(
+                    work_tumor_i, "extra_features.tsv")
+                if not os.path.exists(extra_features_tsv) or restart:
+                    extend_features(filtered_vcf,
+                                    ensemble_beds[
+                                        i] if ensemble_tsv else None,
+                                    extra_features_tsv,
+                                    reference, tumor_bam, normal_bam,
+                                    min_mapq, snp_min_bq,
+                                    dbsnp, None,
+                                    num_threads)
+                extra_features_bed = os.path.join(
+                    work_dataset_split, "extra_features.bed")
+                if not os.path.exists(extra_features_bed) or restart:
+                    extract_ensemble(extra_features_tsv,
+                                     extra_features_bed, True)
+                if ensemble_tsv:
+                    merged_features_bed = os.path.join(
+                        work_dataset_split, "merged_features.bed")
+                    if not os.path.exists(merged_features_bed) or restart:
+                        exclude_ens_variants = []
+                        with open(merged_features_bed, "w") as o_f:
```

Comment:
@msahraeian you can combine both `with` instead of nesting one within the other.